packages/sdk/src/conversation/sso/ssoExecutionDialog.ts

```diff
@@ -0,0 +1,307 @@
+import { ErrorCode, ErrorMessage, ErrorWithCode } from "../../core/errors";
+import { internalLogger } from "../../util/logger";
+
+let DIALOG_NAME = "SsoExecutionDialog";
+let TEAMS_SSO_PROMPT_ID = "TeamsFxSsoPrompt";
+let COMMAND_ROUTE_DIALOG = "CommandRouteDialog";
+
+/**
+ * Sso execution dialog, use to handle sso command
+ */
+export class SsoExecutionDialog extends ComponentDialog {
+  private dedupStorage: Storage;
+  private dedupStorageKeys: string[] = [];
+  private commandMapping: Map<string, string | RegExp | (string | RegExp)[]> = new Map<
+    string,
+    string | RegExp | (string | RegExp)[]
+  >();
+
+  /**
+   * Creates a new instance of the SsoExecutionDialog.
+   * @param dedupStorage Helper storage to remove duplicated messages
+   * @param settings The list of scopes for which the token will have access
+   * @param teamsfx {@link TeamsFx} instance for authentication
+   */
+  constructor(
+    dedupStorage: Storage,
+    ssoPromptSettings: TeamsBotSsoPromptSettings,
+    teamsfx: TeamsFx,
+    dialogName?: string
+  ) {
+    super(dialogName ?? DIALOG_NAME);
+    if (dialogName) {
+      DIALOG_NAME = dialogName;
+      TEAMS_SSO_PROMPT_ID = dialogName + TEAMS_SSO_PROMPT_ID;
+      COMMAND_ROUTE_DIALOG = dialogName + COMMAND_ROUTE_DIALOG;
+    }
+
+    this.initialDialogId = COMMAND_ROUTE_DIALOG;
+
+    this.dedupStorage = dedupStorage;
+    this.dedupStorageKeys = [];
+
+    const ssoDialog = new TeamsBotSsoPrompt(teamsfx, TEAMS_SSO_PROMPT_ID, ssoPromptSettings);
+    this.addDialog(ssoDialog);
+
+    const commandRouteDialog = new WaterfallDialog(COMMAND_ROUTE_DIALOG, [
+      this.commandRouteStep.bind(this),
+    ]);
+    this.addDialog(commandRouteDialog);
+  }
+
+  /**
+   * Add TeamsFxBotSsoCommandHandler instance
+   * @param handler {@link SsoExecutionDialogHandler} callback function
+   * @param triggerPatterns The trigger pattern
+   */
+  public addCommand(handler: SsoExecutionDialogHandler, triggerPatterns: TriggerPatterns): void {
+    const commandId = uuidv4();
+    const dialog = new WaterfallDialog(commandId, [
+      this.ssoStep.bind(this),
+      this.dedupStep.bind(this),
+      async (stepContext: any) => {
+        const tokenResponse: TeamsBotSsoPromptTokenResponse = stepContext.result.tokenResponse;
+        const context: TurnContext = stepContext.context;
+        const message: CommandMessage = stepContext.result.message;
+
+        try {
+          if (tokenResponse) {
+            handler(context, tokenResponse, message);
+          } else {
+            await context.sendActivity("Failed to retrieve user token from conversation context.");
+          }
+          return await stepContext.endDialog();
+        } catch (error: unknown) {
+          await context.sendActivity("Failed to retrieve user token from conversation context.");
+          await context.sendActivity((error as Error).message as string);
+          return await stepContext.endDialog();
+        }
+      },
+    ]);
```

Comment:
may give `shouldTrigger` another name to help people better understand the code. At the beginning I thought we checks whether we need to execute the command here again. Actually we only want to get the result of regex capture group.

Reply:
OK